### ISSUE TITLE
Local authentication is broken

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#895]
Following up on the pylint issue reported in cim_http.py(721):
```
E:726,30: Non-iterable value local_auth_header is used in an iterating context (not-an-iterable)
```
revealed a larger issue, namely that the code path for local authentication for OpenWBEM and OpenPegasus can never be reached, and the credential-based Basic authentication is always used.

From cim_http.py, ca. line 721:
```
    if local_auth_header is not None:
        # The following pylint stmt disables a false positive, see
        # https://github.com/PyCQA/pylint/issues/701
        # TODO 3/16 AM: Track resolution of this Pylint bug.
        # pylint: disable=not-an-iterable
        header_tuples.append(*local_auth_header)
    elif creds is not None:
        auth = '%s:%s' % (creds[0], creds[1])
        auth64 = _ensure_unicode(base64.b64encode(
            _ensure_bytes(auth))).replace('\n', '')
        header_tuples.append(('Authorization', 'Basic %s' % auth64))
    elif locallogin is not None:
        header_tuples.append(('PegasusAuthorization',
                              'Local "%s"' % locallogin))
```
The code above is *outside* of the retry loop in which a returned HTTP status 401 is handled automatically for some cases by setting `local_auth_header`, so the condition on the first `if` statement never becomes true.

In addition, the code contains a bug in that `if`-statement in the statement:
```
header_tuples.append(*local_auth_header)
```
because `local_auth_header` is set to a tuple(header_name, header_value), so the line shown above provides append() with two arguments, resulting in TypeError being raised.

The first bug was introduced in PR #366 (which went into pywbem 0.9.0), and because we test at a higher level in pywbem, it was not found. The second bug has always been there. Looking into our test coverage reveals that this line is in fact never executed, e.g. results from a recent run: https://coveralls.io/builds/14694937/source?filename=pywbem%2Fcim_http.py#L723

There is also some good news about this: It seems that nobody uses local authentication, because the second bug has always been there (I checked back to 0.7.0, i.e. https://github.com/pywbem/pywbem/blob/ab9ef9a332e7e32d02c7cc2861ab0f6e2c05d04c/trunk/cim_http.py#L186).

**Comment by andy-maier** [comment]
Addressed in PR #896.

**Comment by andy-maier** [comment]
We discussed this and Karl wants to find out how the local auth works for Pegasus.
